app/views/viewMovement/ViewMovementView.scala.html

```diff
@@ -66,24 +66,7 @@
         <div class="govuk-grid-row">
             <div class="govuk-grid-column-one-half" id="actions">
                 @h2(messages("viewMovement.actions.heading"), "govuk-heading-l")
-
-                @if(isConsignor) {
-                    @list(Seq(
-                        link(appConfig.emcsTfeCancelMovementUrl(ern, arc), "viewMovement.cancelMovement", Some("cancel-this-movement"), hintKey = Some("viewMovement.cancelMovement.info")),
-                        link(appConfig.emcsTfeChangeDestinationUrl(ern, arc), "viewMovement.changeDestination", Some("submit-a-change-of-destination"), hintKey = Some("viewMovement.changeDestination.info")),
-                        link(appConfig.emcsTfeExplainDelayUrl(ern, arc), "viewMovement.explainDelay", Some("explain-a-delay"), hintKey = Some("viewMovement.explainDelay.info")),
-                        link(appConfig.emcsTfeExplainShortageOrExcessUrl(ern, arc), "viewMovement.explainShortageOrExcess", Some("explain-shortage-or-excess"), hintKey = Some("viewMovement.explainShortageOrExcess.info")),
-                        link("print", "viewMovement.printOrSaveEad", Some("print-or-save-ead"), hintKey = Some("viewMovement.printOrSaveEad.info"))
-                    ), extraClasses = Some("govuk-list--spaced"))
-                } else {
-                    @list(Seq(
-                        link(appConfig.emcsTfeAlertOrRejectionUrl(ern, arc), "viewMovement.alertOrRejection", Some("submit-alert-or-rejection"), hintKey = Some("viewMovement.alertOrRejection.info")),
-                        link(appConfig.emcsTfeReportAReceiptUrl(ern, arc), "viewMovement.reportAReceipt", Some("submit-report-of-receipt"), hintKey = Some("viewMovement.reportAReceipt.info")),
-                        link(appConfig.emcsTfeExplainDelayUrl(ern, arc), "viewMovement.explainDelay", Some("explain-a-delay"), hintKey = Some("viewMovement.explainDelay.info")),
-                        link(appConfig.emcsTfeExplainShortageOrExcessUrl(ern, arc), "viewMovement.explainShortageOrExcess", Some("explain-shortage-or-excess"), hintKey = Some("viewMovement.explainShortageOrExcess.info")),
-                        link("print", "viewMovement.printOrSaveEad", Some("print-or-save-ead"), hintKey = Some("viewMovement.printOrSaveEad.info"))
-                    ), extraClasses = Some("govuk-list--spaced"))
-                }
+                @{actionLinksBody}
```

Comment:
This is good that this has been moved out to a Helper, much neater and easier to test.

Only thinking would be what I said eariler that I don't really like passing HTML into a view to render. I think it might be nice to dependency inject the helper to the view and then call the method on the helper directly from within the view. That way, the view is responsible for the rendering and presentation of HTML without the Controller having to pass HTML to it.

Reply:
https://github.com/hmrc/emcs-tfe-frontend/commit/96f2ca11c67fbc7c5ed23734bdfeea0e60191157